kms/asymmetric/samples.go

```
@@ -0,0 +1,189 @@
+	}
+	message := fmt.Sprintf("%s", messageArr)
+	return message, nil
+}
+
+// [END kms_decrypt_rsa]
+
+// [START kms_encrypt_rsa]
+// Encrypt plaintext message using saved RSA public key
+func encryptRSA(client *cloudkms.Service, ctx context.Context, message, keyPath string) (string, error) {
+	abstractKey, err := getAsymmetricPublicKey(client, ctx, keyPath)
+	if err != nil {
+		return "", err
+	}
+
+	//perform type assertion to get rsa key
+	rsaKey := abstractKey.(*rsa.PublicKey)
+
+	ciphertextBytes, err := rsa.EncryptOAEP(sha256.New(), rand.Reader, rsaKey, []byte(message), nil)
+	if err != nil {
+		return "", err
+	}
+	ciphertextStr := base64.StdEncoding.EncodeToString(ciphertextBytes)
+	return ciphertextStr, nil
+}
+
+// [END kms_encrypt_rsa]
+
+// [START kms_sign_asymmetric]
+// Sign a message using saved asymmetric private key
+func signAsymmetric(client *cloudkms.Service, ctx context.Context, message, keyPath string) (string, error) {
+	//find hash of message
+	digest := sha256.New()
+	digest.Write([]byte(message))
+	digestStr := base64.StdEncoding.EncodeToString(digest.Sum(nil))
+
+	asymmetricSignRequest := &cloudkms.AsymmetricSignRequest{
+		Digest: &cloudkms.Digest{
+			Sha256: digestStr,
+		},
+	}
+
+	response, err := client.Projects.Locations.KeyRings.CryptoKeys.CryptoKeyVersions.
+		AsymmetricSign(keyPath, asymmetricSignRequest).Context(ctx).Do()
+	if err != nil {
+		return "", err
+	}
+
+	return response.Signature, nil
+}
+
+// [END kms_sign_asymmetric]
+
+// [START kms_verify_signature_rsa]
+// Verify the cryptographic signature for a message signed with an RSA private key
+func verifySignatureRSA(client *cloudkms.Service, ctx context.Context, signature, message, keyPath string) error {
+	abstractKey, err := getAsymmetricPublicKey(client, ctx, keyPath)
+	if err != nil {
+		return err
+	}
+	//perform type assertion to get rsa key
+	rsaKey := abstractKey.(*rsa.PublicKey)
+	decodedSignature, _ := base64.StdEncoding.DecodeString(signature)
```

dont ignore this error